libbeat/processors/actions/decode_base64_field.go

```diff
@@ -84,55 +77,55 @@ func (f *decodeBase64Field) Run(event *beat.Event) (*beat.Event, error) {
 		backup = event.Fields.Clone()
 	}
 
-	err := f.decodeField(f.config.Field.From, f.config.Field.To, event.Fields)
-	if err != nil && f.config.FailOnError {
+	err := f.decodeField(event)
+	if err != nil {
 		errMsg := fmt.Errorf("failed to decode base64 fields in processor: %v", err)
-		f.log.Debug("decode base64", errMsg.Error())
-		event.Fields = backup
-		_, _ = event.PutValue("error.message", errMsg.Error())
-		return event, err
+		f.log.Debug(errMsg.Error())
+		if f.config.FailOnError {
+			event.Fields = backup
+			event.PutValue("error.message", errMsg.Error())
+			return event, err
+		}
 	}
-
 	return event, nil
 }
 
 func (f decodeBase64Field) String() string {
 	return fmt.Sprintf("%s=%+v", processorName, f.config.Field)
 }
 
-func (f *decodeBase64Field) decodeField(from string, to string, fields common.MapStr) error {
-	value, err := fields.GetValue(from)
+func (f *decodeBase64Field) decodeField(event *beat.Event) error {
+	value, err := event.GetValue(f.config.Field.From)
 	if err != nil {
-		// Ignore ErrKeyNotFound errors
 		if f.config.IgnoreMissing && errors.Cause(err) == common.ErrKeyNotFound {
 			return nil
 		}
-		return fmt.Errorf("could not fetch value for key: %s, Error: %s", from, err)
+		return fmt.Errorf("could not fetch base64 value for key: %s, Error: %v", f.config.Field.From, err)
 	}
 
-	text, ok := value.(string)
-	if !ok {
+	base64String, isString := value.(string)
+	if !isString {
 		return fmt.Errorf("invalid type for `from`, expecting a string received %T", value)
 	}
 
-	decodedData, err := base64.StdEncoding.DecodeString(text)
+	decodedData, err := base64.StdEncoding.DecodeString(base64String)
 	if err != nil {
-		return fmt.Errorf("error trying to unmarshal %s: %v", text, err)
+		return fmt.Errorf("error trying to decode %s: %v", base64String, err)
 	}
 
-	field := to
+	target := f.config.Field.To
 	// If to is empty
-	if to == "" || from == to {
+	if f.config.Field.To == "" || f.config.Field.From == f.config.Field.To {
 		// Deletion must happen first to support cases where a becomes a.b
-		if err = fields.Delete(from); err != nil {
```

Comment:
Do you mind removing this `if`? It's not required in this processor, but got copied from rename processor.

Reply:
done